Src/Lib/ResourceHelper.cpp

```diff
@@ -476,8 +476,19 @@ HBITMAP LoadImageFile( const wchar_t *path, const SIZE *pSize, bool bUseAlpha, b
 			frameWidthD=abs(pSize->cx);
 			if (pSize->cy)
 				frameHeightD=pSize->cy;
+			else if (maxSize && maxSize->cy > 0)
```

Comment:
I'd recommend to move this block outside of `if (pSize->cy)` block.
We are computing `frameHeightD` inside original _if_/_else_ block and then if the computed value exceeds provided `maxSize` we will just adjust.

```
if (maxSize && maxSize->cy > 0 && frameHeightD > maxSize->cy)
{
    ...
}
```